galley/pkg/config/analysis/local/analyze.go

```diff
@@ -42,36 +44,40 @@ var (
 	apiserverNew = apiserver.New
 )
 
-// SourceAnalyzer handles local analysis of k8s and file based event sources
+// SourceAnalyzer handles local analysis of k8s event sources, both live and file-based
 type SourceAnalyzer struct {
 	m                    *schema.Metadata
 	sources              []event.Source
 	analyzer             analysis.Analyzer
 	transformerProviders transformer.Providers
 
-	// Which collections are used by this analysis
-	// Derived from the specified analyzer and transformer providers
-	inputCollections map[collection.Name]struct{}
+	// Which kube resources are used by this analysis
+	// Derived from metadata and the specified analyzer and transformer providers
+	kubeResources schema.KubeResources
 
+	// Hook function called when a collection is used in analysis
 	collectionReporter snapshotter.CollectionReporterFn
 }
 
 // NewSourceAnalyzer creates a new SourceAnalyzer with no sources. Use the Add*Source methods to add sources in ascending precedence order,
 // then execute Analyze to perform the analysis
-func NewSourceAnalyzer(m *schema.Metadata, analyzer analysis.Analyzer, cr snapshotter.CollectionReporterFn) *SourceAnalyzer {
+func NewSourceAnalyzer(m *schema.Metadata, analyzer analysis.Analyzer, cr snapshotter.CollectionReporterFn, serviceDiscovery bool) *SourceAnalyzer {
```

Comment:
Nit: Aggregating some of the parameters into a config object may make sense here.

Reply:
I played with this a bit but couldn't find something that felt cleaner. Putting some params into an Options object made the function call shorter, but ended up outweighed by the extra boilerplate around it.

I'm open to more specific suggestions if you have them, otherwise if it's not a strong objection I'd just as soon move on.